sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/EventProcessorHost.cs

```diff
@@ -190,26 +191,92 @@ public sealed class EventProcessorHost
+                eventHubPath,
+                consumerGroupName,
+                tokenProvider,
+                cloudStorageAccount,
+                leaseContainerName,
+                storageBlobPrefix,
+                operationTimeout,
+                transportType)
+        {
+        }
+
+        /// <summary>
+        /// Create a new host to process events from an Event Hub with provided <see cref="TokenProvider"/>
+        /// </summary>
+        /// <param name="hostName">Name of the processor host. MUST BE UNIQUE. Strongly recommend including a Guid to ensure uniqueness.</param>
+        /// <param name="endpointAddress">Fully qualified domain name for Event Hubs. Most likely, {yournamespace}.servicebus.windows.net</param>
+        /// <param name="eventHubPath">The name of the EventHub.</param>
+        /// <param name="consumerGroupName">The name of the consumer group within the Event Hub.</param>
+        /// <param name="tokenProvider">Token provider which will generate security tokens for authorization.</param>
+        /// <param name="checkpointManager">Object implementing ICheckpointManager which handles partition checkpointing.</param>
+        /// <param name="leaseManager">Object implementing ILeaseManager which handles leases for partitions.</param>
+        /// <param name="operationTimeout">Operation timeout for Event Hubs operations.</param>
+        /// <param name="transportType">Transport type on connection.</param>
+        public EventProcessorHost(
+            string hostName,
+            Uri endpointAddress,
+            string eventHubPath,
+            string consumerGroupName,
+            ITokenProvider tokenProvider,
+            ICheckpointManager checkpointManager,
+            ILeaseManager leaseManager,
+            TimeSpan? operationTimeout = null,
+            TransportType transportType = TransportType.Amqp)
         {
             Guard.ArgumentNotNullOrWhiteSpace(nameof(hostName), hostName);
             Guard.ArgumentNotNull(nameof(endpointAddress), endpointAddress);
             Guard.ArgumentNotNullOrWhiteSpace(nameof(eventHubPath), eventHubPath);
             Guard.ArgumentNotNullOrWhiteSpace(nameof(consumerGroupName), consumerGroupName);
             Guard.ArgumentNotNull(nameof(tokenProvider), tokenProvider);
-            Guard.ArgumentNotNull(nameof(cloudStorageAccount), cloudStorageAccount);
-            Guard.ArgumentNotNullOrWhiteSpace(nameof(leaseContainerName), leaseContainerName);
+            Guard.ArgumentNotNull(nameof(checkpointManager), checkpointManager);
+            Guard.ArgumentNotNull(nameof(leaseManager), leaseManager);
 
             this.HostName = hostName;
             this.EndpointAddress = endpointAddress;
             this.EventHubPath = eventHubPath;
             this.ConsumerGroupName = consumerGroupName;
-            this.OperationTimeout = operationTimeout ?? ClientConstants.DefaultOperationTimeout;
-            this.TransportType = TransportType;
```

Comment:
Surprised to see this has always been wrongly its own `TransportType` instead of calling the one passed to the constructor.